src/ray/protobuf/autoscaler_state_service.proto

```diff
@@ -0,0 +1,173 @@
+// ============= Cluster Resources ====================
+//
+// Following fields represents the Cluster Resources autoscaler interested
+// in.
+
+// Represents an anti-affinity constraint. A bundle with this constraint
+// can't be allocated to a node that has a label with the same name and
+// value.
+message AntiAffinityConstraint {
+  string label_name = 1;
+  string label_value = 2;
+}
+
+message ResourceBundle {
+  map<string, double> resources = 1;
+  repeated AntiAffinityConstraint affinity_constraints = 2;
+  // If the resource bundle is allocated, the node will be labeled
+  // with the following labels.
+  map<string, string> new_labels = 3;
```

Comment:
TO adaptor node affinity with labels,  I suggest the following two options:  
The current new_labels and affinity_constraints cannot fully express node affinity, and the following expressions are required.

**Options 1:**
```
message ResourceBundle {
  map<string, double> resources = 1;
  repeated LabelMatchExpression node_affinity_expressions = 2;
}

enum LabelMatchOperator {
  IN = 0;
  NOT_IN = 1;
  EXISTS = 2;
  DOES_NOT_EXIST = 3;
}

message LabelMatchExpression {
  string key = 1;
  LabelMatchOperator label_match_operator = 2;
  repeated string values = 3;
  bool soft = 4;
}

```


**Options 2:**
```
message ResourceBundle {
  map<string, double> resources = 1;
  SchedulingStrategy scheduling_strategy = 2;
}

enum LabelMatchOperator {
  IN = 0;
  NOT_IN = 1;
  EXISTS = 2;
  DOES_NOT_EXIST = 3;
}

message LabelMatchExpression {
  string key = 1;
  LabelMatchOperator label_match_operator= 2;
  repeated string values = 3;
  bool soft = 4;
}

message NodeAffinitySchedulingStrategy {
  repeated LabelMatchExpression match_expressions = 1;
}

message SchedulingStrategy {
  oneof scheduling_strategy {
    // Default hybrid scheduling strategy.
    DefaultSchedulingStrategy default_scheduling_strategy = 1;
    // Placement group based scheduling strategy.
    PlacementGroupSchedulingStrategy placement_group_scheduling_strategy = 2;
    // Best effort spread scheduling strategy.
    SpreadSchedulingStrategy spread_scheduling_strategy = 3;
    NodeAffinitySchedulingStrategy node_affinity_scheduling_strategy = 4;
  }
}
```